### ISSUE TITLE
Font Library: Implement audible feedback for actions that only have visual feedback

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#58319]
### Description

Any non-native action that only has visual feedback must be accompanied with an audibel feedback for screen reader users.

Native actions such as, for example, selection of an item within a native `<select>` element already provide native accessibility.

Author-created actions aren't native. As such, the missing accessibility must be implemented via authoring.

In the Fonts modal dialog, examples of these actions are:
- Navigation to the 'sub views' with the fonts details and back. This is not a 'native' navigation to a new resource, it's just rendering new content. When a screen reader users click a font, they won't have a clue what happened on the page.
- Search results / noresults must be announced. Ideally, any visual and audible feedback should be debounced untile users stoop typing.
- All the notices when installing / deleting fonts should already handle internally audibel feedback via `speak` messages.
- Any other visual update / feedback and such must be announced audibly.

### Step-by-step reproduction instructions

As an example:
- Use a screen reader for example on macOS use Safari and VoiceOver.
- Go to Site editor > Styles > Typography > Manage fonts.
- Go to the Google fonts tab.
- Search for some term that returns some search results.
- Observe that the search results aren't announced.
- Search for some dummy tern that doesn't return any result.
- Observe that there is no audible feedback about the search returning no results.

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes
